modules/ln_glgui/glgui.scm

```diff
@@ -230,12 +230,43 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ;; process an input event
 ;; 20100519: allow multiple guis
 ;; 20100804: support gui offset
-(define (glgui-event guis t x0 y0)
-  (if (and glgui:active app:width app:height)
-    (let ((gs (if (list? guis) guis (list guis))))
-      (if (fx= t EVENT_REDRAW) (apply glgui:render gs)
-        (apply glgui:inputloop (append (list t x0 y0) gs)))
-   )))
+
+(define glgui-frame-period-max (make-parameter 0.5)) ;; How long to sleep at most in redraw.
+
+;; glgui-wakeup! - a thunk, which when called will immediately unblock
+;; the thread waiting in glgui-event.  Should be called if other
+;; threads notice the event loop should proceed.
+(define glgui-wakeup!)
+
+(define glgui-event
+  (let ((consecutive-redraw-count 1)
```

Comment:
If you were to change this to 0.05, and in L254 change to `(set! consecutive-redraw-count (fl+ consecutive-redraw-count step))` and change the reset in L249 to `step`, we could eliminate the multiplication in L251. It could even be changed to `0.0` in which case a redraw after an input even would be immediate.